CRM/Contact/Form/Task.php

```diff
@@ -190,9 +190,7 @@ public static function preProcessCommon(&$form, $useTable = FALSE) {
         // and it decides when to use distinct based on input criteria, which needs
         // to be fixed and optimized.
 
-        foreach ($allCids[$cacheKey] as $cid => $ignore) {
-          $form->_contactIds[] = $cid;
-        }
+        $form->_contactIds = array_unique($allCids[$cacheKey]);
```

Comment:
This was my first attempt to fix the issue, but I guess it is nod needed. I will revert this.